ni_measurementlink_generator/ni_measurementlink_generator/templates/_helpers.py

```diff
@@ -0,0 +1,313 @@
+"""Functions to set up and tear down sessions of NI devices in NI TestStand."""
+
+import ni_measurementlink_service as nims
+import nidcpower
+import nidmm
+import nifgen
+import niscope
+import niswitch
+
+from ni_measurementlink_generator.templates.internal._internal import (
+    GrpcChannelPoolHelper,
+    PinMapClient,
+)
+
+
+def update_pin_map(pin_map_id: str) -> None:
+    """Update registered pin map contents.
+
+    Create and register a pin map if a pin map resource for the specified pin map id is not found.
+
+    Args:
+        pin_map_id (str):
+            The resource id of the pin map to register as a pin map resource. By
+            convention, the pin_map_id is the .pinmap file path.
+
+    """
+    with GrpcChannelPoolHelper() as grpc_channel_pool:
+        pin_map_client = PinMapClient(grpc_channel=grpc_channel_pool.pin_map_channel)
+        pin_map_client.update_pin_map(pin_map_id)
+
+
+def create_nidcpower_sessions(pin_map_id: str) -> None:
```

Comment:
I can see why these methods need to be different since the session types are distinct for the drivers, but is there not any way we can share common code between all these by parameterizing things in common function calls?

Reply:
Yes Joel. I am working on updating the create and destroy sessions method to take in instrument type as a parameter.

Reply:
FYI, I refactored the NI-Digital version: https://github.com/ni/measurementlink-python/pull/245